.github/workflows/test.yml

```diff
@@ -0,0 +1,34 @@
+name: Test
+
+# This workflow will run automatically on every PR against `main` branch. It can be also trigerred
+# manually via the Github UI.
+on:
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: ./.github/actions/setup
```

Comment:
In `widgets`, we had both `actions/checkout` and this one, chained one after another. I don't think that is necessary, since former is already included.